client/src/Annotator/annotator.hooks.tsx

```diff
@@ -0,0 +1,290 @@
+            setMetadata(data.image.metadata || {});
+            setFilename(data.image.file_name);
+            setNext(data.image.next);
+            setPrevious(data.image.previous);
+            setCategoriesIds(data.image.category_ids || []);
+            setAnnotating(data.image.annotating || []);
+
+            setPreferences(data.preferences || {});
+
+            setDataset(data.dataset);
+        };
+
+        try {
+            addProcess(dispatch, process);
+            update();
+        } catch (error) {
+            // TODO display toast
+        } finally {
+            removeProcess(dispatch, process);
+        }
+    }, [imageId, dispatch]);
+
+    return {
+        dataset,
+        categories,
+        metadata,
+        filename,
+        next,
+        previous,
+        categoriesIds,
+        annotating,
+        preferences,
+    };
+};
+
+// Initialize Paper.js and Canvas with image
+export const useCanvas = (imageUrl: string): CanvasState => {
+    const paperRef = useRef<Maybe<paper.PaperScope>>(null);
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+    const rasterRef = useRef<Maybe<paper.Raster>>(null);
+
+    const [imgScale, setImgScale] = useState<number>(0); // TODO
+    const [imgData, setImgData] = useState<Maybe<ImageData>>(null); // TODO
+    const [rasterSize, setRasterSize] = useState<Maybe<RasterSize>>(null);
+
+    const centerImageAction = useCallback(() => {
+        if (!paperRef.current || !canvasRef.current || !rasterRef.current) {
+            return;
+        }
+
+        const parentX = rasterRef.current.width || 0;
+        const parentY = rasterRef.current.height || 0;
+
+        paperRef.current.view.zoom = Math.min(
+            (canvasRef.current.width / parentX) * 0.95,
+            (canvasRef.current.height / parentY) * 0.8,
+        );
+
+        setImgScale(1 / paperRef.current.view.getZoom());
+        paperRef.current.view.setCenter(0, 0);
+    }, [canvasRef, rasterRef]);
+
+    const onWheelAction = useCallback((e: WheelEvent<HTMLDivElement>) => {
+        if (!paperRef.current || !paperRef.current.view) return;
+
+        const view = paperRef.current.view;
+
+        if (!view.center) return;
+
+        const centerPt = view.center;
+
+        if (e.altKey) {
+            // pan up and down
+            const delta = new paper.Point(0, 0.5 * e.deltaY);
+            const newCenterP = centerPt.add(delta);
+            paperRef.current.view.setCenter(newCenterP); // TODO
+        } else if (e.shiftKey) {
+            // pan left and right
+            const delta = new paper.Point(0.5 * e.deltaY, 0);
+            const newCenterP = centerPt.add(delta);
+            paperRef.current.view.setCenter(newCenterP);
+        } else {
+            // changeZoom
+            const viewPosition = view.viewToProject(
+                new paper.Point(e.nativeEvent.offsetX, e.nativeEvent.offsetY),
+            );
+
+            const factor = 1.2;
```

Comment:
since this ins't changing it should be moved out of the scope of the function to the module scope and named with big letters only.